app/scripts/modules/core/src/application/applications.component.ts

```diff
@@ -0,0 +1,150 @@
+
+import { ACCOUNT_SERVICE, AccountService, IAccount } from 'core/account/account.service';
+import { ANY_FIELD_FILTER } from 'core/presentation/anyFieldFilter/anyField.filter';
+import { APPLICATION_READ_SERVICE, ApplicationReader, IApplicationSummary } from 'core/application/service/application.read.service';
+import { Application } from 'core/application';
+import { ICache } from 'core/cache/deckCache.service';
+import { OVERRIDE_REGISTRY, OverrideRegistry } from 'core/overrideRegistry/override.registry';
+import { VIEW_STATE_CACHE_SERVICE, ViewStateCacheService } from 'core/cache/viewStateCache.service';
+
+import './applications.less';
+
+export interface IApplicationPagination {
+  currentPage: number;
+  itemsPerPage: number;
+  maxSize: number;
+}
+
+export class ApplicationsController implements IComponentController {
+  private accounts: IAccount[];
+  public applications: IApplicationSummary[];
+  private applicationsViewStateCache: ICache;
+  public filteredApplications: IApplicationSummary[];
+  public menuActions: any;
+  public applicationsLoaded = false;
+  public pagination = this.getDefaultPagination();
+
+  public constructor(
+    private $scope: IScope,
+    private $uibModal: IModalService,
+    private accountService: AccountService,
+    private $filter: IFilterService,
+    private $state: StateService,
+    private applicationReader: ApplicationReader,
+    private viewStateCache: ViewStateCacheService,
+    private overrideRegistry: OverrideRegistry) {
+    'ngInject';
+
+    this.applicationsViewStateCache = this.viewStateCache.get('applications') || this.viewStateCache.createCache('applications', { version: 1 });
+
+    this.accountService.listAccounts().then((accounts) => {
+      this.accounts = accounts;
+    });
+
+    this.menuActions = [
+      {
+        displayName: 'Create Application',
+        action: () => {
+          this.$uibModal.open({
+            scope: this.$scope,
+            templateUrl: this.overrideRegistry.getTemplate('createApplicationModal', require('./modal/newapplication.html')),
+            controller: this.overrideRegistry.getController('CreateApplicationModalCtrl'),
+            controllerAs: 'newAppModal'
+          }).result.then((app) => this.routeToApplication(app));
+        }
+      }
+    ];
+
+    this.applicationReader.listApplications().then((applications) => {
+      applications.forEach((app) => this.fixAccount(app));
+      this.applications = applications;
+      this.filterApplications();
+      this.applicationsLoaded = true;
+    });
+
+    this.$scope.$watch('viewState', () => this.cacheViewState(), true);
+
+    this.initializeViewState();
+  }
+
+  private cacheViewState(): void {
+    this.applicationsViewStateCache.put('#global', this.$scope.viewState);
+  }
+
+  private initializeViewState(): void {
+    this.$scope.viewState = this.applicationsViewStateCache.get('#global') || {
+        sortModel: { key: 'name' },
+        applicationFilter: '',
+      };
```

Comment:
would be really nice to move this off the scope (I understand why it's there, maybe include a TODO to add handlers to the fields in the UI to get the scope out)